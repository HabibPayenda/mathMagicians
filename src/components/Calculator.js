import './styles.css';

import React, { useState } from 'react';
import Header from './Header';
import Main from './Main';
import Button from './Button';
import Footer from './Footer';
import buttons from './constants/buttons';
import footerButtons from './constants/footerButtons';
import calculate from '../functions/calculate';

const Calculator = () => {
  const [total, setTotal] = useState(0);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const obj = {
    total,
    next,
    operation,
  };

  const changeCalculation = (text) => {
    const calculation = calculate(obj, text);
    const { total, next, operation } = calculation;
    setTotal(total);
    setNext(next);
    setOperation(operation);
  };

  const btns = buttons.map((item) => {
    const button = (
      <Button
        large={item.size}
        color={item.color}
        text={item.text}
        key={item.text}
        changeCalculation={(text) => changeCalculation(text)}
      />
    );
    return button;
  });

  const footerBtns = footerButtons.map((item) => {
    const button = (
      <Button
        large={item.size}
        color={item.color}
        text={item.text}
        key={item.text}
        changeCalculation={changeCalculation}
      />
    );
    return button;
  });

  return (
    <div className="container">
      <Header total={total} next={next} operation={operation} />
      <Main>{btns}</Main>
      <Footer>{footerBtns}</Footer>
    </div>
  );
};

export default Calculator;
