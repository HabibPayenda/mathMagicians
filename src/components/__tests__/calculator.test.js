import '@testing-library/jest-dom';
import reactTestRenderer from 'react-test-renderer';

import Calculator from '../Calculator';

describe('Calculator', () => {
  it('should match the snapshot', () => {
    const tree = reactTestRenderer.create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });
});
