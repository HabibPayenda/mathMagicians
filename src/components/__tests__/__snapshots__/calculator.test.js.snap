// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="header"
    data-testid="header"
  >
    0
  </div>
  <div
    className="mainContainer"
  >
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      AC
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      +/-
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      %
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      /
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      7
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      8
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      9
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      x
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      4
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      5
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      6
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      -
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      3
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      +
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="button gray large"
      data-testid="button"
      onClick={[Function]}
    >
      0
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      .
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      =
    </div>
  </div>
</div>
`;

exports[`should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="header"
    data-testid="header"
  >
    0
  </div>
  <div
    className="mainContainer"
  >
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      AC
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      +/-
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      %
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      /
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      7
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      8
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      9
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      x
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      4
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      5
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      6
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      -
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      3
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      +
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="button gray large"
      data-testid="button"
      onClick={[Function]}
    >
      0
    </div>
    <div
      className="button gray small"
      data-testid="button"
      onClick={[Function]}
    >
      .
    </div>
    <div
      className="button orange small"
      data-testid="button"
      onClick={[Function]}
    >
      =
    </div>
  </div>
</div>
`;
